routeros/resource_system_certificate_acme.go

```diff
@@ -0,0 +1,294 @@
+package routeros
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+// https://help.mikrotik.com/docs/display/ROS/Certificates
+// https://wiki.mikrotik.com/wiki/Manual:System/Certificates
+func ResourceSystemCertificateAcme() *schema.Resource {
+	resSchema := map[string]*schema.Schema{
+		MetaResourcePath: PropResourcePath("/certificate"),
+		MetaId:           PropId(Id),
+		MetaTransformSet: PropTransformSet("challenge_type:type"),
+
+		// ACME:
+		"eab_hmac_key": {
+			Type:             schema.TypeString,
+			Optional:         true,
+			ForceNew:         true,
+			Sensitive:        true,
+			Description:      "HMAC key for ACME External Account Binding (optional).",
+			DiffSuppressFunc: AlwaysPresentNotUserProvided,
+		},
+		"eab_kid": {
+			Type:             schema.TypeString,
+			Optional:         true,
+			ForceNew:         true,
+			Description:      "Key identifier.",
+			DiffSuppressFunc: AlwaysPresentNotUserProvided,
+		},
+		"directory_url": {
+			Type:             schema.TypeString,
+			Optional:         true,
+			ForceNew:         true,
+			Description:      "ACME directory url.",
+			DiffSuppressFunc: AlwaysPresentNotUserProvided,
+		},
+		"dns_name": {
+			Type:     schema.TypeString,
+			Optional: true,
+			ForceNew: true,
+			Description: " If the dns-name is not specified, it will default to the automatically generated " +
+				"`/ip cloud name` (ie. http://example.sn.mynetname.net).",
+			DiffSuppressFunc: AlwaysPresentNotUserProvided,
+		},
+		"challenge_type": {
+			Type:             schema.TypeString,
+			Optional:         true,
+			ForceNew:         true,
+			Description:      "ACME challenge.",
+			ValidateFunc:     validation.StringInSlice([]string{"cloud-dns", "http-01"}, false),
+			DiffSuppressFunc: AlwaysPresentNotUserProvided,
+		},
+
+		// Certificate:
+		"akid": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Authority Key Identifier.",
+		},
+		"common_name": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Common Name (e.g. server FQDN or YOUR name).",
+		},
```

Comment:
Since this is now separate from `resource_system_certificate`, maybe it makes sense to keep only one of `dns_name` and `common_name`, since they are always equal for ACME certs?